google/cloud/pubsub/samples/samples.cc

```diff
@@ -1052,9 +1053,9 @@ void OptimisticSubscribe(std::vector<std::string> const& argv) {
         std::move(response->handler).ack();
         return gc::Status();
       }
-      if (response.status().code() == gc::StatusCode::kUnavailable &&
-          response.status().message() == "no messages returned") {
-        std::cout << "No messages returned from Pull()\n";
+      if (gc::internal::IsClientOrigin(response.status())) {
```

Comment:
we shouldn't recommend internal APIs in samples. Maybe just leave the sample as it was.

Reply:
I remember. Either `IsClientOrigin` needs to be public, or we say the sample is fine how it is.

Also, FTR, I still think we should attach the client origin metadata, regardless of whether we check for it in the sample.

Reply:
Done